### PR TITLE
Add PR finalization agent

### PR DESCRIPTION
**Pull request by sgraband**
#### What it does
Introduces ChatAgent to assist users in creating PRs based on current changes.
The system prompt now asks for the git diff, status, ticket link, and PR template.
Added two variables, #git-status and #git-diff, for capturing status and diff information.
Added the "Invoke PR finalization" command, which opens the chat and sends a predefined message including the git diff and status.
Includes various bug fixes.

Based on #122.

#### How to test
- Create some changes in your current workspace
Either:
- Ask the chat to help you with creating a PR (feel free to use the `#git-diff `and `#git-status` variables when asked)
or:
- Invoke the "Invoke PR finalization" command

#### Follow-ups

Currently this hardcodes the agent of the chat to the new agent. This should be removed before merging. Once #51 is merged, we should refactor this to take those changes into account.

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/ai-chat/src/common/chat-service.ts [hunk 1]
```diff
@@ -143,7 +149,9 @@ export class ChatServiceImpl implements ChatService {
         const chatAgents = this.chatAgentService.getAgents();
         if (chatAgents.length > 0) {
             // TODO collect the correct agent
-            this.chatAgentService.invokeAgent(chatAgents[0].id, requestModel);
+            // FIXME: This is a temporary solution to get the PrFinalization agent (Issue #6)
```
Comment (tortmayr) [inline packages/ai-chat/src/common/chat-service.ts:152]: Agent selection is now merged in main. You can remove this and should be able to select your agent by using "@PrFinalization" at the beginning of your prompt

File: packages/ai-pr-finalization/src/browser/git-commands.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            execute: () => this.getGitDiff()
+        });
+    }
+
+    protected getGitStatus(): Promise<string> {
+        return this.getGitOutput('status');
+    }
+
+    protected getGitDiff(): Promise<string> {
+        return this.getGitOutput('diff');
```
Comment (tortmayr) [inline packages/ai-pr-finalization/src/browser/git-commands.ts:43]: The #git-diff varible seems to fail if there is nothing to diff. E.g. just adding a new file will result in an empty git diff.
If I then invoke the `Invoke AI Finalization" command the chat bot never responds (keeps generating)

**Review by tortmayr — COMMENTED** [review]
Looks good to me. I only have to minor inline comments

File: packages/ai-pr-finalization/src/browser/git-commands.ts (hunk 2 above)
Comment (lucas-koehler) [reply packages/ai-pr-finalization/src/browser/git-commands.ts:43]: Good catch, thanks! I fixed it in https://github.com/eclipsesource/theia/pull/124/commits/48f7b36a8a5d4c86fb8b416617ed33a092447205

**Review by tortmayr — APPROVED** [review]
LGTM! 👍🏼

**Pull request merged**